test-integration/helpers/test-element.coffee

```diff
@@ -48,26 +51,37 @@ class TestItemHelper
 
   isPresent: (args...) =>
     locator = @getLocator(args...)
-    @test.driver.isElementPresent(locator).then (isPresent) ->
-      isPresent
+    @test.driver.isElementPresent(locator)
+
+  isDisplayed: (args...) =>
+    @isPresent(args...).then (isPresent) =>
+      if isPresent
+        locator = @getLocator(args...)
+        el = @findElement(args...)
+        el.isDisplayed()
+      else
+        false
 
   # Helper for the common case of `get(...).click()`.
   # Plus, it allows a place to add logging since this is one of the most
   # common places for Selenium to time out (trying to click on an element)
   click: (args...) =>
     locator = @getLocator(args...)
+    # Scroll to the element so it is visible before clicking (this assumes `position: fixed` is overridden for all element)
+    # @isDisplayed(args...).then (isDisplayed) =>
+    #   unless isDisplayed
+    #     el = @findElement(args...)
+    #     @test.utils.windowPosition.scrollTo(el)
     @test.utils.verboseWrap "Clicking #{JSON.stringify(locator)}", =>
-      @get(args...).click()
+      el = @get(args...)
+      @test.utils.windowPosition.scrollTo(el)
+      el.click()
 
-# Using defined properties for access eliminates the possibility
-# of accidental assignment
-Object.defineProperties TestItemHelper.prototype,
-  test:
-    get: -> @_test
-  locator:
-    get: -> @_locator
-  element:
-    get: -> @get()
+  waitClick: (args...) =>
+    locator = @getLocator(args...)
+    @test.utils.verbose "Waiting for #{JSON.stringify(locator)}"
+    @test.utils.wait.for(locator)
+    @click(args...)
 
```

Comment:
hmm...is that case common enough? seems like it would be just as quick to do
```coffee
blahBlah.el.thing().findElement().click()
```
but i guess it would nice to have an immediate one to match the `waitClick`.